packages/data/addon/adapters/elide-facts.ts

```diff
@@ -62,50 +55,62 @@ export default class ElideFacts extends EmberObject implements NaviFactAdapter {
    * @param options
    * @returns Promise that resolves to the result of the AsyncQuery creation mutation
    */
-  createAsyncQueryRequest(request: RequestV1, options: RequestOptions): Promise<AsyncQueryResponse> {
+  createAsyncQuery(request: RequestV1, options: RequestOptions = {}): Promise<AsyncQueryResponse> {
     const mutation: DocumentNode = GQLQueries['asyncFactsMutation'];
-    const dataQuery = this.dataQueryFromRequest(request);
-    const clientId: string = options.clientId || v1();
+    const query = this.dataQueryFromRequest(request);
+    const id: string = options.requestId || v1();
 
     // TODO: Add other options based on RequestOptions
-    const queryOptions: ApolloMutationOptions = {
-      mutation,
-      variables: {
-        ids: [clientId],
-        query: dataQuery
-      }
-    };
-
+    const queryOptions = { mutation, variables: { id, query } };
     return this.apollo.mutate(queryOptions);
   }
 
   /**
    * @param id
-   * @returns Promise with the updated asyncquery's id and status
+   * @returns Promise with the updated asyncQuery's id and status
    */
-  cancelAsyncRequest(id: string) {
+  cancelAsyncQuery(id: string) {
     const mutation: DocumentNode = GQLQueries['asyncFactsCancel'];
-    return this.apollo.mutate({
-      mutation,
-      variables: {
-        ids: [id]
-      }
-    });
+    return this.apollo.mutate({ mutation, variables: { id } });
+  }
+
+  /**
+   * @param id
+   * @returns Promise that resolves to the result of the AsyncQuery fetch query
+   */
+  fetchAsyncQuery(id: string) {
+    const query: DocumentNode = GQLQueries['asyncFactsQuery'];
+    return this.apollo.query({ query, variables: { ids: [id] } });
   }
 
   /**
    * @param _request
    * @param _options
    */
   urlForFindQuery(_request: RequestV1, _options: RequestOptions): string {
-    throw new Error('Method not implemented.');
+    return 'TODO';
   }
 
   /**
-   * @param _request
-   * @param _options
+   * @param request
+   * @param options
    */
-  fetchDataForRequest(_request: RequestV1, _options: RequestOptions): Promise<TODO> {
-    throw new Error('Method not implemented.');
+  @task(function*(this: ElideFacts, request: RequestV1, options: RequestOptions) {
+    let asyncQueryPayload = yield this.createAsyncQuery(request, options);
+    const asyncQuery = asyncQueryPayload?.asyncQuery.edges[0]?.node;
+    const { id } = asyncQuery;
+    let status: QueryStatus = asyncQuery.status;
+
+    while (status === QueryStatus.QUEUED || status === QueryStatus.PROCESSING) {
+      yield timeout(this._pollingInterval);
+      asyncQueryPayload = yield this.fetchAsyncQuery(id);
+      status = asyncQueryPayload?.asyncQuery.edges[0]?.node.status;
+    }
+    return asyncQueryPayload;
+  })
+  fetchDataForRequestTask!: TODO;
```

Comment:
```suggestion
  fetchDataForRequestTask!: Promise<AsyncQueryResponse>;
```

Reply:
This is a task, this is probably could go on `fetchDataForRequest`